src/operator/combineLatest.ts

```diff
@@ -31,3 +31,28 @@ export function combineLatest<T, R>(...observables: Array<Observable<any> |
 
   return new ArrayObservable(observables).lift<T, R>(new CombineLatestOperator<T, R>(project));
 }
+
+/* These types would be come standard types used through-out the library */
+export type ObservableOrPromise<T> = Observable<T> | Promise<T>;
+export type ArrayOrIterator<T> = Iterator<T> | ArrayLike<T> | Array<T>;
+export type ObservableInput<T> = Observable<T> | Promise<T> | Iterator<T> | ArrayLike<T>;
+
+/* tslint:disable:max-line-length */
+export interface CombineLatestDeclaration<T> {
```

Comment:
Naming is hard, I just picked something at the time that didn't clash.

I kind of like `CombineLatestSignature` it's descriptive as to what this interface really is about.